docs/knowledgebase/solution-cannot-perform-ondat-operations-to-volumes-or-nodes.md

```diff
@@ -0,0 +1,101 @@
+```
+
+- Volumes are not successfully failing over.
+- Nodes cannot be added or deleted.
+
+In addition, when reviewing the logs for the Ondat daemonset, you may see the following `error` and `warning` messages;
+
+```bash
+# Get the logs for the Ondat daemonset/
+kubectl logs daemonsets.apps/storageos-node  --namespace=storageos
+
+# truncated output...
+{"auth_req_username":"storageos","error":"context deadline exceeded","level":"error","msg":"error while performing login","req_id":"4ba51f16-0f69-4b4f-bb83-0d7f9b58ff0c","req_ip":"10.73.0.217:49118","req_xff":"","time":"2022-02-13T18:48:54.796090712Z"}
+{"error":"context deadline exceeded","level":"error","msg":"storing remote capacity failed","node_id":"d8d3d8dc-c3cd-4f97-86fb-4706d01dff33","time":"2022-02-13T18:49:03.674273759Z"}
+{"error":"context deadline exceeded","level":"warning","msg":"failed to update node size value in remote store","node_id":"d8d3d8dc-c3cd-4f97-86fb-4706d01dff33","time":"2022-02-13T18:49:03.674370036Z"}
+{"error":"context deadline exceeded","level":"error","msg":"failed to fetch latest available licence information","time":"2022-02-13T18:49:03.675980929Z"}
+{"error":"context deadline exceeded","level":"warning","msg":"lock operation failure - is etcd healthy?","node_id":"d8d3d8dc-c3cd-4f97-86fb-4706d01dff33","store_lock_key":"storageos/default/v1/locks/node/d8d3d8dc-c3cd-4f97-86fb-4706d01dff33","time":"2022-02-13T18:49:11.026773582Z"}
+{"error":"context deadline exceeded","level":"error","msg":"abandoning expired lock","node_id":"d8d3d8dc-c3cd-4f97-86fb-4706d01dff33","store_lock_key":"storageos/default/v1/locks/node/d8d3d8dc-c3cd-4f97-86fb-4706d01dff33","time":"2022-02-13T18:49:11.026865788Z"}
+
+# Use jq for clearer formatting and only show the error messages in the daemonset pods.
+kubectl logs daemonsets.apps/storageos-node  --namespace=storageos | jq '{msg: .msg, error: .error}'
+
+{
+"msg": "storing remote capacity failed",
+"error": "context deadline exceeded"
+}
+{
+"msg": "failed to update node size value in remote store",
+"error": "context deadline exceeded"
+}
+{
+"msg": "failed to fetch latest available licence information",
+"error": "context deadline exceeded"
+}
+{
+"msg": "lock operation failure - is etcd healthy?",
+"error": "context deadline exceeded"
+}
+{
+"msg": "abandoning expired lock",
+"error": "context deadline exceeded"
+}
+{
+"msg": "error during authentication",
+"error": "context deadline exceeded"
+}
+```
+
+## Resolution
+
+- To resolve the operations issues, this will require assessing and repairing the  `etcd`  cluster and ensuring that the cluster is healthy again - whether the issue is `etcd`, your network configuration, or because the nodes where Ondat is deployed are overutilised to the point where they cannot fulfil requests to and from `etcd` successfully.
+
+1. Ensure that your `etcd` cluster is running and is healthy.
+    1. `etcdctl member list -wtable`
```

Comment:
maybe change these to bullets (like -, -, -) so the numbering wouldn't be confusing?

Reply:
fixed